components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-service-provider.jsp

```diff
@@ -87,6 +87,16 @@ location.href = "list-service-providers.jsp";
 
     if (samlIssuerName != null && "update".equals(action)){
     	appBean.setSAMLIssuer(samlIssuerName);
+
+        // Inbound authentication components might have set an application certificate in the session.
+        // One usage in this scenario is, using the certificate inside SAML SP metadata.
+        String applicationCertificate = (String) session.getAttribute("applicationCertificate");
```

Comment:
Is this being set into the session. And what's the need of storing this in the session. I couldn't find the place where it's being set. Looks like it should be read from the request parameter and set into the bean.

Reply:
This is needed to handle honoring the embedded certificate in a SAML metadata file. The certificates stored along with the service provider itself are read from the request parameters.